logical/logical.go

```diff
@@ -47,6 +64,17 @@ type Backend interface {
 	// to the backend. The backend can use this to clear any caches or reset
 	// internal state as needed.
 	InvalidateKey(key string)
+
+	// Configure is used to set up the backend based on the provided backend
+	// configuration. This is primarily used in the the context of backend
+	// plugins, where the RPC server has no context of the core's config.
+	// This is the equivalent of calling framework.Backend.Setup(conf), but
+	// from the RPC client to configure components such as the logger, storage,
+	// system view, etc.
+	Configure(*BackendConfig) error
```

Comment:
Sorry, at this point I'm confused :-)

It sounds like the nicest thing to do might be to just codify what framework has done as a function in the `logical.Backend` interface:

```go
Setup(*logical.BackendConfig) (logical.Backend, error)
```

Does this work for you?